multiview/embed/gcca.py

```diff
@@ -0,0 +1,251 @@
+
+        Parameters
+        ----------
+        X : array-like, shape (n_observations, n_features)
+            The data to preprocess
+
+        Returns
+        -------
+        X2 : preprocessed data matrix
+        """
+
+        # Mean along rows using sample mean and sample std
+        X2 = stats.zscore(X, axis=1, ddof=1)
+        # Mean along columns
+        mu = np.mean(X2, axis=0)
+        X2 -= mu
+        return X2
+
+    def _check_inputs(self, fraction_var, sv_tolerance, n_components, size):
+        if fraction_var is None:
+            pass
+        elif not isinstance(fraction_var, float) and not isinstance(
+            fraction_var, int
+        ):
+            raise TypeError("fraction_var must be an integer or float")
+        elif fraction_var <= 0 or fraction_var > 1:
+            raise ValueError("fraction_var must be in (0,1]")
+
+        if sv_tolerance is None:
+            pass
+        elif not isinstance(sv_tolerance, int):
+            raise TypeError("sv_tolerance must be an integer")
+        elif sv_tolerance <= 0:
+            raise ValueError("sv_tolerance must be greater than 0")
+
+        if n_components is None:
+            pass
+        elif not isinstance(n_components, int):
+            raise TypeError("n_components must be an integer")
+        elif n_components <= 0:
+            raise ValueError("n_components must be greater than 0")
+        elif n_components > min(size):
+            raise ValueError(
+                "n_components must be less than or equal to the \
+                    minimum input rank"
+            )
+
+    def fit(
+        self,
+        Xs,
+        fraction_var=0.9,
+        sv_tolerance=None,
+        n_components=None,
+        tall=False,
+    ):
+        """
+        Calculates a projection from each view to a latentent space such that
+        the sum of pariwise latent space correlations is maximized. Each view
+        'X' is normalized and the left singular vectors of 'X^T X' are
+        calculated using SVD. The number of singular vectors kept is determined
+        by either the percent variance explained, a given rank threshold, or a
+        given number of components. The singular vectors kept are concatenated
+        and SVD of that is taken and used to calculated projects for each view.
+
+        Parameters
+        ----------
+        Xs: list of array-likes - Xs shape: (n_views,) - Xs[i] shape:
+            (n_samples, n_features_i) The data to fit to. Each sample will
+            receive its own embedding.
+        fraction_var : percent, default=0.9
+            Explained variance for rank selection during initial SVD of each
+            sample.
+        sv_tolerance : float, optional, default=None Singular value
+            threshold for rank selection during initial SVD of each sample.
+        n_components : int (positive), optional, default=None Rank to
+            truncate to during initial SVD of each sample. tall : boolean,
```

Comment:
Looks like an autoformatter is messing up your comments. tall is on wrong line again